web/src/stream_edit.js

```diff
@@ -535,6 +535,11 @@ export function initialize() {
     // This handler isn't part of the normal edit interface; it's the convenient
     // checkmark in the subscriber list.
     $("#streams_overlay_container").on("click", ".sub_unsub_button", (e) => {
+        if ($(e.currentTarget).hasClass("disabled")) {
+            // We do not allow users to subscribe themselves to private streams.
+            return;
```

Comment:
Since we do not use `button` element here, we cannot add `disabled` attribute here to not allow clicking on it and we can also not set `pointer-events: none` as we need to show the tooltip. So, added this code to avoid making API request when clicking on "+" for private streams.